specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,136 @@
+
+# Background
+
+This is the spec for proposal [Windows App Runtime Deployment API #1240](https://github.com/microsoft/WindowsAppSDK/issues/1240).
+
+Windows App SDK (WAS) deployment consists of a Main package, multiple Framework packages, and for
+unpackaged apps a set of Dynamic Dependency packages, all of which are published by Microsoft 
+and are Store-Signed. Consumers of WAS will be both Microsoft and non-Microsoft publishers.
+In order to have WAS fully present and functional, the WAS Framework and Main packages must be 
+deployed to the system. While the Windows Application Model naturally supports framework
+dependencies it does not have a means of declaring a dependency on Main packages. This API spec
+and feature  proposal is to provide a means of identifying when the WAS dependencies are not
+satisfied, and if that is the case, provide a means to resolve the missing dependency. 
+
+The key scenario that is expected to have a missing dependency is for packaged MSIX which have
+declared the WAS frameworks as a dependency. In this scenario it is likely the main package would
+be missing because they cannot declare a dependency on the main package and it will not be
+deployed by the Store. For MSIX packages with permissions, this dependency could be immediately
+resolved online or offline if the package size is small and the framework carries the dependency.
+
+A class of MSIX packages that are Full Trust or have package management capabilities, such as
+Centennial MSIX packages will have the permissions to immediately resolve the dependency. These
+packages are the target for this version of the API. These packages will have the capability
+to deploy packages and satisfy the dependency. Since the Main Package for WAS is expected to be
+very small, this API has a streamlined solution for this scenario, which is to carry the WAS
+main packages inside the WAS framework, and leverage the full trust capability of the package
+to deploy the main package. A later release will address how to resolve the missing packages
+for UWPs when they become supported.
+
+
+# Description
+
+When a Windows App SDK (WAS) consuming Centennial package is deployed from the Store, it is
+expected to only have the App package and frameworks, and the WAS main package may not be present,
+or may not be at the desired version. This API enables a developer to detect when the required WAS
+main and/or framework packages are not present and take corrective action.
+
+Since this API is part of the WAS framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
```

Comment:
You usually say Main as a proper noun meaning the WAS Main package, and main as in a package where PackageType=Main (not any particular package). You should do likewise for Framework/framework.

You'll probably find it helpful to use WARmain, WARsingleton, WARfwk and WARsingleton like https://github.com/microsoft/WindowsAppSDK/blob/main/specs/Deployment/MSIXPackageVersioning.md#123-windows-app-sdks-msix-packages. I found a similar pain writing that spec -- writing out the full formal WinAppSDK package names _degraded_ clarity. You could up top cite the MSIXPackageVersioning.md spec for the list of WinAppSDK's MSIX packages and how you're using that terminology through the rest of this spec. Then you could say `WARfwk` here instead of `the WAS Framework` and likewise throughout the spec.

As a perk you reduce the need to update this spec in the future if/when we add (or remove) a package, since you just refer to that spec for the exact package list except where noteworthy to cite a specific package or subset.